event-exporter/sinks/stackdriver/monitored_resource_factory_test.go

```diff
@@ -0,0 +1,152 @@
+package stackdriver
+
+import (
+	"testing"
+
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+var anyEvent = &api_v1.Event{}
+
+func TestDefaultMonitoredResourceType(t *testing.T) {
+	tests := []struct {
```

Comment:
I think you can have even simpler test with only one method, for example test parameters would be: resource model (old or new), event and expected monitored resource (and you just verify actual vs expected)